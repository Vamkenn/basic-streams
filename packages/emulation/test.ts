import {EventsList, emulate, t, v} from "./index"

expect.addSnapshotSerializer(EventsList.jestSerializer)

describe("emulate", () => {
  test("map", () => {
    expect(
      emulate(create => {
        return cb => create(t(10), v(1))(x => cb(x + 1))
      }),
    ).toMatchSnapshot()
  })

  test("filter", () => {
    expect(
      emulate(create => {
        return cb =>
          create(v(1), t(10), v(0), t(10), v(1))(x => {
            if (x > 0) cb(x)
          })
      }),
    ).toMatchSnapshot()
  })

  test("order of simultaneous values", () => {
    expect(
      emulate(create => {
        const a = create(t(10), v(1))
        const b = create(t(10), v(2))
        return cb => {
          const disposeA = a(cb)
          const disposeB = b(cb)
          return () => {
            disposeA()
            disposeB()
          }
        }
      }),
    ).toMatchSnapshot()
  })

  test("unsubscribe", () => {
    expect(
      emulate(create => {
        const a = create(t(10), v(1), t(10), v(0))
        const b = create(t(15), v(null))
        return cb => b(a(cb))
      }),
    ).toMatchSnapshot()
  })

  test("unsubscribe doesn't remove other values with same cb", () => {
    expect(
      emulate(create => {
        const a = create(t(10), v(1), t(10), v(0))
        const b = create(t(10), v(1), t(10), v(0))
        const unsubscriber = create(t(15), v(null))
        return cb => {
          a(cb)
          unsubscriber(b(cb))
          return () => {}
        }
      }),
    ).toMatchSnapshot()
  })
})
