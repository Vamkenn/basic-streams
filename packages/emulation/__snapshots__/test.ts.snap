// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emulate filter 1`] = `
EventsList(
  0: 1
  20: 1
)
`;

exports[`emulate map 1`] = `
EventsList(
  10: 2
)
`;

exports[`emulate order of simultaneous values 1`] = `
EventsList(
  10: 1
  10: 2
)
`;

exports[`emulate unsubscribe 1`] = `
EventsList(
  10: 1
)
`;

exports[`emulate unsubscribe doesn't remove other values with same cb 1`] = `
EventsList(
  10: 1
  10: 1
  20: 0
)
`;
